### ISSUE TITLE
Graphic buttons whose name is different from the label in the legend - violation of 2.5.3?

### ISSUE THREAD

**Issue by JAWS-test** [w3c/wcag#1260]
I have a web application with buttons whose label consists only of a graphic. However, there is a legend at the bottom of the page explaining the purpose of the graphic (e.g. "Edit" for the pen, "Search" for the magnifying glass). Now, however, the label in the legend differs from the Accessible Name of the button (via aria-label). Is this a violation of SC 2.5.3? Because it is not against the wording of the SC, but against the [purpose](https://www.w3.org/WAI/WCAG21/Understanding/label-in-name.html#benefits)

**Comment by guyhickling** [comment]
I would say yes, since those words at the foot of the page are "the text that is presented visually" to quote the SC. If I were auditing that site I would fail it. Dragon users would try to speak the words Edit and Search, but if they aren't in the accessible names from their ARIA labels, then it won't work for them. 

(Having said that, it wouldn't work for them anyway even without the help hints at the foot of the page, as Edit and Search are the words skilled Dragon users would be likely to try speaking anyway for those two icons, and its hard to think what other words they might try speaking for them!)

Out of interest, what are the ARIA label texts?

**Comment by alastc** [comment]
Not 2.5.3, but it appears to be a 3.2.4: Consistent Identification issue.

**Comment by patrickhlauke** [comment]
not if they're consistently (mis)labelled. whether their accessible name is different from what the legend elsewhere says they are is not directly a 3.2.4 issue I'd say...

**Comment by guyhickling** [comment]
3.2.4 is about components that are repeated across a set of pages. The question is about one page so 3.2.4 doesn't apply, unless I'm missing something?

**Comment by JAWS-test** [comment]
In my case, SC 3.2.4 would not apply in any case, since they are all consistently labeled:
* Consistent for AT users with a name in the aria label
* consistent for sighted users with a graphic that has an explanation in the legend

Problem was that label in the legend differs from the name in the aria-label.

**Comment by fstrr** [comment]
This issue looks like it can be closed. If it needs to be re-opened, please do that and convert it to a Discussion.
